wallet/src/main/scala/org/bitcoins/wallet/LockedWallet.scala

```diff
@@ -314,6 +314,19 @@ abstract class LockedWallet extends LockedWalletApi with BitcoinSLogger {
     }
   }
 
+  override def getAddressInfo(
+      address: BitcoinAddress): Future[Option[AddressInfo]] = {
```

Comment:
Thoughts on flattening this to just a `Future`? How explicitly do you expect users to need to deal with the `None` case very differently from the `Future.failed` case?

Reply:
My reasoning here is twofold: 

1) A failed future isn't encoded into the type, which is bad. In that way its similar to `Option.get`
2) A failed future might be triggered by a DB problem, a network problem, a file system problem, or a bunch of things. Whereas a `None` cleanly indicates that the address isn't in the wallet

Reply:
I do agree with this reasoning, but it does make the api harder to work with with `for` expressions

Reply:
I would argue that `Future[T]` is just a wrapper for `Option[Try[T]]` but that point aside, I think that if the future is failed with a descriptive error then the issue of distinguishing a missing address from a db problem isn't an issue

Reply:
Use case for having this return `Future[Option[..]]`: 

A user wants to detect if an address is _not_ in the wallet. With your approach the user would then either match on the type of failure in the future (which he gets no help from the compiler doing, as futures aren't parameterized on the error type), or check if the error message on the exception matches a certain substring (which is bad)

Reply:
I would say there's calls on Futures like `transform` and `onComplete` and `recover` that would let you handle this nicely if the error used in failure has its own type specific to a missing address but at the same time I don't think this is a huge deal so if you still think that `Future[Option]` is the way to go I'm okay with going with it and having the discussion if need be when it gets used in the future